File: c++/Sem -1/leap_year.cpp
#include <iostream>
using namespace std;
int main()
{
    int a = 0;
    cout << "Enter the year : ";
    cin >> a;
    if (a % 4 == 0)
    {
        if (a % 100 == 0)
        {
            if (a % 400 == 0)
            {

                cout << "the year is leap year.";
            }
            else
            {

                cout << "The year is not leap year.";
            }
        }
        else
        {
            cout << "The year is leap year.";
        }
    }
    else
    {
        cout << "The year is not leap year.";
    }
    return 0;
}

File: Python/leap_year.py
def is_leap(year):
    leap = False
    # Write your logic here
    if 1900 <= year and year < 100000:
        if year % 4 == 0 and year % 100 == 0 and year % 400 == 0:
            leap = True
            return leap
        elif year % 4 == 0 and year % 100 == 0:
            leap = False
            return leap
        elif year % 4 == 0:
            leap = True
            return leap
        elif year % 400 == 0:
            leap = True
            return leap
        else:
            return leap
    else:
        return leap


year = int(input())
print(is_leap(year))
